### PR TITLE
Use new API of zope.testbrowser >= 5.0.0

### PR DESCRIPTION
**Pull request by florianpilz**

### CODE REVIEW

**Comment by florianpilz** [comment]
You are right, users of `z3c.etestbrowser` have to restructure their tests. Major version bump is required.

**Comment by florianpilz** [comment]
@mgedmin Will you do it? Working on different project this week.

**Comment by mgedmin** [comment]
Unfortunately I'm also busy.

**Comment by icemac** [comment]
@florianpilz There is already [z3c.etestbrowser.wsgi.Browser](https://github.com/zopefoundation/z3c.etestbrowser/blob/master/src/z3c/etestbrowser/wsgi.py). The implementation of `z3c.etestbrowser.testing. ExtendedTestBrowser` looks quite similar. Maybe `wsgi.py`should import from `testing.py`.

**Comment by wteuber** [comment]
@florianpilz is this PR worth keeping?

**Comment by icemac** [comment]
Rebasing onto master.

**Comment by icemac** [comment]
@mgedmin Do you like these changes?

File: src/z3c/etestbrowser/wsgi.txt [hunk 1]
```diff
@@ -1,8 +1,9 @@
...
+Deprecated special support for zope.testbrowser.wsgi
+----------------------------------------------------
 
-There is also a variant in ``z3c.etestbrowser.wsgi`` which can be used for
-the WSGI variant of ``zope.testbrowser``.
+There was also a variant in ``z3c.etestbrowser.wsgi`` which could be used for
+the WSGI variant of ``zope.testbrowser``. It is no longer necessary because.
+``z3c.etestbrowser.testing`` now speaks WSGI.
 
 Example:
```
Comment (sallner) [inline src/z3c/etestbrowser/wsgi.txt]: This basically tests the deprecation, which is fine, and will go with 4.0. I wonder, whether we should place a hint somewhere to remove that, after the 3.0 Release.

File: src/z3c/etestbrowser/wsgi.py [hunk 2]
```diff
@@ -11,20 +11,12 @@
...
-    Based on zope.testbrowser.wsgi.Browser.
-
-    """
-
-
-# Just for backwards compatibility with versions before 2.0:
-ExtendedTestBrowser = Browser
+zope.deferredimport.deprecated(
+    "z3c.etestbrowser.testing.ExtendedTestBrowser now speaks WSGI, so please "
+    "use it. This BBB import will go away in Version 4.0.",
```
Comment (mgedmin) [inline src/z3c/etestbrowser/wsgi.py]: I don't think "version" needs to be capitalized.

File: src/z3c/etestbrowser/wsgi.txt (hunk 1 above)
Comment (icemac) [reply src/z3c/etestbrowser/wsgi.txt]: I created #13 and will add a note about the planned removal.

File: src/z3c/etestbrowser/wsgi.py (hunk 2 above)
Comment (icemac) [reply src/z3c/etestbrowser/wsgi.py]: Indeed!

**Comment by florianpilz** [comment]
Great to see this finally merged. 😄 Did repeatedly think about  closing this PR in the last 2½ years, but always hoped it would be picked up so the Python 3 dream lives on.

Party

**Comment by mgedmin** [comment]
Thank you for not giving up!  We are sorry for things taking so long...

**Comment by florianpilz** [comment]
No need to be sorry, you did the work to finish up. Long live Zope!

**Pull request merged**